src/keydown_content.ts

```diff
@@ -1,119 +0,0 @@
-/** Shim for the keyboard API because it won't hit in FF57. */
-
-import * as Messaging from "./messaging"
-import * as msgsafe from "./msgsafe"
-import { isTextEditable, getAllDocumentFrames } from "./dom"
-import { isSimpleKey } from "./keyseq"
-
-function keyeventHandler(ke: KeyboardEvent) {
-    // Ignore JS-generated events for security reasons.
-    if (!ke.isTrusted) return
-
-    // Mode is changed based on ke target in the bg.
-    modeSpecificSuppression(ke)
-
-    Messaging.message("keydown_background", "recvEvent", [
-        msgsafe.KeyboardEvent(ke),
-    ])
-}
-
-// {{{ Bad key suppression system
-
-// This is all awful and will go away when we move the parsers and stuff to content properly.
```

Comment:
I think I'm just beginning to realise how awful this was - did we _really_ have two parsers running at the same time just so we could suppress the keys?